core/state/trie_prefetcher.go

```diff
@@ -253,40 +239,24 @@ func (sf *subfetcher) schedule(keys [][]byte) {
 	sf.lock.Lock()
 	sf.tasks = append(sf.tasks, keys...)
 	sf.lock.Unlock()
-
-	// Notify the prefetcher, it's fine if it's already terminated
-	select {
-	case sf.wake <- struct{}{}:
-	default:
-	}
-}
-
-// peek tries to retrieve a deep copy of the fetcher's trie in whatever form it
-// is currently.
-func (sf *subfetcher) peek() Trie {
-	ch := make(chan Trie)
-	select {
-	case sf.copy <- ch:
-		// Subfetcher still alive, return copy from it
-		return <-ch
-
-	case <-sf.term:
-		// Subfetcher already terminated, return a copy directly
-		if sf.trie == nil {
-			return nil
-		}
-		return sf.db.CopyTrie(sf.trie)
-	}
+	// Notify the prefetcher. The wake-chan is buffered, so this is async.
+	sf.wake <- true
 }
 
-// abort interrupts the subfetcher immediately. It is safe to call abort multiple
+// wait waits for the subfetcher to finish it's task. It is safe to call wait multiple
 // times but it is not thread safe.
-func (sf *subfetcher) abort() {
-	select {
-	case <-sf.stop:
-	default:
-		close(sf.stop)
+func (sf *subfetcher) wait() {
+	// Signal termination by nil tasks
+	sf.lock.Lock()
+	if sf.closing {
+		return // already exiting
 	}
+	sf.closing = true
+	sf.tasks = nil
```

Comment:
We cannot set it to `nil` here, then we risk losing items. We must only every `nil` it in the `loop`